### PR TITLE
Resolve MethodSource parameters from Kotlin Companion Object

### PR DESCRIPTION
**Pull request by bedla**
## Overview

Hi,
I found that when I write Parameterized tests in Kotlin with `@MethodSource` referred method from Kotlin class have to be placed in Companion object and marked with `@JvmStatic` annotation.
This PR make it possible to use regular companion object's method.
What do you think?
Thx,
Ivos

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ x [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

**Comment by marcphilipp** [comment]
Are you aware that using `@TestInstance(Lifecycle.PER_CLASS)` let's you use regular test class methods (not in the companion object)?

**Comment by bedla** [comment]
I understand usage of `@TestInstance(Lifecycle.PER_CLASS)` as workaround. 
I will put tests for this case.

**Comment by bedla** [comment]
Hi, 
I have fixed ITs from `platform-tooling-support-tests` (it was hard to run it correctly on local Windows machine directly from IntelliJ IDEA) and updated documentation with code examples for Kotlin.
Ivos

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -12,5 +12,5 @@ dependencies {
 	implementation("de.marcphilipp.gradle:nexus-publish-plugin:0.4.0")
 	implementation("biz.aQute.bnd:biz.aQute.bnd.gradle:5.2.0")
 	implementation("com.github.jengelman.gradle.plugins:shadow:6.1.0")
-	implementation("org.gradle:test-retry-gradle-plugin:1.1.8")
+	implementation("org.gradle:test-retry-gradle-plugin:1.1.9")
```
Comment (bedla) [inline buildSrc/build.gradle.kts:15]: I have updated this dependency because of tests were failing on old ASM version that is not able to read bytecode version 59

File: platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/JavacModulesTests.java [hunk 2]
```diff
@@ -120,6 +120,8 @@
 				args.add("--patch-module");
 				args.add(project.moduleName + "=" + path);
 			}
+			args.add("-encoding");
```
Comment (bedla) [inline platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/JavacModulesTests.java:123]: I have added explicit encoding because tests were failing on my local Windows with Czech locale (CP-1250)

File: platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/ModularUserGuideTests.java [hunk 3]
```diff
@@ -94,6 +94,8 @@
 					// Don't include command-line tools that "require io.github.classgraph"
 					.filter(s -> !s.contains("tools")).forEach(args::add);
 		}
+		args.add("-encoding");
```
Comment (bedla) [inline platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/ModularUserGuideTests.java:97]: same as JavacModulesTests.java
